### PR TITLE
Added code for generating BST from it's Preorder Traversal #2605

### PR DESCRIPTION
**Pull request by sakshigoel19**
### Have you read the [Contributing Guidelines on Pull Requests] (https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
 Yes

### Description

Given preorder traversal of a given BST, generate the binary search tree and return it's root.

### Checklist

- [Yes] I've read the contribution guidelines. 
- [Yes] I've checked the issue list before deciding what to submit. 
- [Yes] I've edited the `README.md` and link to my code. 

## Related Issues or Pull Requests

#2605

### CODE REVIEW

**Review by ankitaggarwal23 — APPROVED** [review]

File: C-Plus-Plus/ds/BST_from_Preorder_Traversal.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    Node *root = BstFromPreorder(pre, size);  
+    cout<<"Inorder traversal of the constructed tree: \n"; //Inorder traversal printed to view the constructed BST 
+    Inorder(root);  
+    return 0;  
+}
+
+/*
+Time Complexity : O(n)
+Space Complexity : O(n)
+Sample Input:
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/BST_from_Preorder_Traversal.cpp]: Give a new line after SC.

File: C-Plus-Plus/ds/BST_from_Preorder_Traversal.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
+/*
+ Given an array of integers which represents the preorder traversal of a Binary Search Tree
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/BST_from_Preorder_Traversal.cpp]: Remove the space before the word `Given`

File: C-Plus-Plus/ds/BST_from_Preorder_Traversal.cpp [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    return new_node;
+}
+
+//Function to generate BST from preorder traversal 
+//Iterative solution using stack which runs in O(n) time
+Node* BstFromPreorder (int pre[], int size){
+    stack <Node*> stk;
+    Node *root = CreateNode (pre[0]);     //First element of preorder traversal is always the root
+    Node *p = root;                      
+    int count = 1;                        //count maintains number of nodes in tree
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/BST_from_Preorder_Traversal.cpp]: Add comment above the line of code.

File: C-Plus-Plus/ds/BST_from_Preorder_Traversal.cpp [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    int size,pre[30];
+    cout << "Enter the number of nodes in tree: ";
+    cin >> size;   
+    cout << "Enter a valid preorder traversal :";
+    for (int i=0 ; i<size ; i++) 
+    {
+        cin >> pre[i];
+    }
+    Node *root = BstFromPreorder(pre, size);  
+    cout<<"Inorder traversal of the constructed tree: \n"; //Inorder traversal printed to view the constructed BST
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/BST_from_Preorder_Traversal.cpp]: Remove the unnecessary comments.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by sakshigoel19** [comment]
@ankitaggarwal23 I have made the requested changes. Please review it.

File: C-Plus-Plus/ds/BST_from_Preorder_Traversal.cpp [hunk 5]
```diff
@@ -0,0 +1,101 @@
+/*
+Given an array of integers which represents the preorder traversal of a Binary Search Tree 
+Construct the BST and return its root.
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/ds/BST_from_Preorder_Traversal.cpp]: Use multiline comments // for description

File: C-Plus-Plus/README.md [hunk 6]
```diff
@@ -87,6 +87,7 @@
 - [Binary Search Tree to Greater Sum Tree](ds/BST_to_Greater_Sum_Tree.cpp)
 - [Threaded Binary Tree](ds/ThreadedBinaryTree.cpp)
 - [Alternate Merging of nodes of two linked lists](ds/alternate_merging_of_linked_list.cpp)
+- [Generate BST from given preorder traversal](ds/BST_from_Preorder_Traversal.cpp)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/README.md]: Make your title concise like Generate BST from preorder traversal or BST from preorder traversal

**Review by ankitaggarwal23 — COMMENTED** [review]
Rest looks good to merge

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Make the few changes

**Comment by sakshigoel19** [comment]
@ankitaggarwal23 I have made the changes asked.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**